### ISSUE TITLE
Samsung Browser logins based on app instead of URL

### ISSUE THREAD

**Issue by Nojwe** [bitwarden/android#1109]
## Steps To Reproduce

Use Samsung Internet Beta, try to log in to a site for the first time. Unfortunately this bug seems to be inconsistent

## Expected Result

Display logins based on URL, rather than app name.

**Comment by cscharf** [comment]
Hi @Nojwe , we've tested this and are unable to reproduce. Is this still an issue for you?

**Comment by Nojwe** [comment]
Yes. It's still happening. Is there a way I can pull a log file for this?
Happy to deal with ADB or anything that would be useful.

**Comment by cscharf** [comment]
Hi @Nojwe , from our customer success team:

> I just reinstalled the browser and tried almost a dozen websites. Not once did Bitwarden look for the browser package name as opposed to the URI. I'm on v13.0.0.33 of the Samsung Internet Beta browser.

Is there a specific website you're seeing this behavior, and what URL do you have saved in your vault item for that website where you're expecting auto-fill? What version of Samsung Internet Beta browser are you using? What behavior are you actually seeing? Do you have just 1 item in your vault for the site, multiple, which app name (the browser you're using or the site or an app similar to the site, etc.)?

I'm afraid we're simply going to need a lot more details about your operating environment, target site(s) you're experiencing this and the parameters/steps to reproduce including auxiliary data for us to build a reproduction of the issue.

**Comment by Nojwe** [comment]
Hi @cscharf 

Here's what I'm using:
Samsung Galaxy S10, software version G973U1UES4ETJ1, October 1 security patch
Samsung Internet Beta 13.0.1.33, but this has happened on previous versions as well.
BitWarden v2.6.1.

Example behavior:
-go to staples.com
-Bitwarden recognizes the username and password fields, the dropdown to go to my vault appears.
-I unlock Bitwarden with my fingerprint.
-Instead of suggesting matches based on the uri staples.com, it shows matches for com.sec.andrdoid.app.sbrowser.beta

There is only one log in for this account (and all my accounts) in my vault. 

URIs are www.staples.com and androidapp://com.sec.andrdoid.app.sbrowser.beta.

This has happened with walgreens.com, misfit.com, amazon.com, bitwarden.com, newegg.com, verizon.com along with a number of others that I unfortunately haven't noted.

After further use I've found that this behavior happens consistently the first time I try to log into any website after rebooting my phone if I have not opened Bitwarden to access the vault or log into another app. Once I have opened Bitwarden and unlocked my vault (through the app, rather than through the autofill popup in the browser), it typically starts working as expected when logging into a website.

Bitwarden is not having its battery usage optimized.

Please let me know if there is any other information I can provide. Thank you very much for your help, I really appreciate it.

**Comment by mportune-bw** [comment]
I'm able to reproduce this on the S10.  Looks like Samsung Internet Beta is not populating ViewNode's `WebDomain` property, which autofill uses to provide the Uri for lookup.  Without that property, Bitwarden falls back to the app's packageName (com.sec.andrdoid.app.sbrowser.beta).  @cscharf Do we have a contact we can pass this along to?

**Comment by cscharf** [comment]
Thanks @mportune-bw ! I don't think we have a direct contact per-say, but we do have the prior contact list we hit up in regards to the package whitelisting issue, I'll dig that up and try to pass this along.

Can we negate using the package name to auto-fill in this instance or is that fallback useful for another use-case I'm not thinking of?

**Comment by mportune-bw** [comment]
> Can we negate using the package name to auto-fill in this instance

I would have to experiment with allowing Accessibility to take over after Autofill has already kicked in.  That path makes a lot of assumptions though.

**Comment by cscharf** [comment]
> I would have to experiment with allowing Accessibility to take over after Autofill has already kicked in. That path makes a lot of assumptions though.

let's hold off on any action for now, thanks Matt!

**Comment by albertopasqualetto** [comment]
It is the same also with Samsung browser non beta (com.sec.android.app.sbrowser) **when I open a link in gmail** (so it doesn't open the full browser app).
Bitwarden searches a correspondence for com.sec.android.app.sbrowser.

**Comment by douglasparker** [comment]
@cscharf this still happens to me.

I switched back to Android and it shows `com.sec.android...` when autofilling from the Samsung Browser.

This has been a long outstanding issue and was one of the reasons I had switched to Apple for a while.

Switching to another browser doesn't help because it still uses Samsung Browser when apps open a browser in-app (forgot the name for this functionality).

I don't believe this is an issue with 1Password. This really makes autofilling a huge pain because I have to search for the credentials manually in Bitwarden.

This issue has been outstanding for a long time now.

Are there any updates on this? I love Bitwarden but am ready tk throw in the towel and stop recommending it.
